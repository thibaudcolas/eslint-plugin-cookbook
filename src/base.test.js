const path = require("path");
const CLIEngine = require("eslint").CLIEngine;

const base = require("./base");

describe("config", () => {
  it("works", () => {
    const cli = new CLIEngine({
      useEslintrc: false,
      configFile: path.join(__dirname, "base.js"),
    });

    expect(cli.executeOnText("var foo\n")).toMatchObject({
      errorCount: 2,
      warningCount: 0,
    });
  });

  it("semver", () => {
    expect(base).toMatchSnapshot();
  });
});
