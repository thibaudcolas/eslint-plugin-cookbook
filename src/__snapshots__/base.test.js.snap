// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config semver 1`] = `
Object {
  "env": Object {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "airbnb",
    "prettier",
    "prettier/flowtype",
    "prettier/react",
  ],
  "rules": Object {
    "import/no-extraneous-dependencies": Array [
      2,
      Object {
        "devDependencies": Array [
          "test/**",
          "tests/**",
          "spec/**",
          "**/__tests__/**",
          "**/__mocks__/**",
          "test.{js,jsx}",
          "test-*.{js,jsx}",
          "**/*{.,_}{test,spec}.{js,jsx}",
          "**/jest.config.js",
          "**/vue.config.js",
          "**/webpack.config.js",
          "**/webpack.config.*.js",
          "**/rollup.config.js",
          "**/rollup.config.*.js",
          "**/gulpfile.js",
          "**/gulpfile.*.js",
          "**/Gruntfile{,.js}",
          "**/protractor.conf.js",
          "**/protractor.conf.*.js",
          "**/dangerfile.js",
        ],
        "optionalDependencies": false,
      },
    ],
    "jsx-a11y/label-has-associated-control": Array [
      0,
    ],
    "jsx-a11y/label-has-for": Array [
      0,
    ],
    "no-param-reassign": Array [
      2,
      Object {
        "props": false,
      },
    ],
    "no-warning-comments": Array [
      1,
      Object {
        "location": "start",
        "terms": Array [
          "todo",
          "fixme",
          "xxx",
        ],
      },
    ],
    "react/forbid-prop-types": Array [
      2,
      Object {
        "forbid": Array [
          "any",
        ],
      },
    ],
    "react/jsx-filename-extension": Array [
      2,
      Object {
        "extensions": Array [
          ".js",
        ],
      },
    ],
    "react/jsx-no-bind": Array [
      0,
    ],
  },
}
`;
